lib/core/include/scipp/core/sizes.h

```diff
@@ -15,6 +17,7 @@
 namespace scipp::core {
 
 constexpr int32_t NDIM_MAX = 6;
```

Comment:
Rename to something like `NDIM_OPS_MAX`? And please add a comment to explain what the two variables mean.